locals.tf

```
@@ -873,11 +891,12 @@ cloudinit_write_files_common = <<EOT
     }
 
     myrename () {
-      local eth="$1"
-      local eth_connection=$(nmcli -g GENERAL.CONNECTION device show $eth || echo '')
-      nmcli connection modify "$eth_connection" \
-        con-name $eth \
-        connection.interface-name $eth
+        local eth="$1"
+        local eth_connection
+        eth_connection=$(nmcli -g GENERAL.CONNECTION device show "$eth" || echo '')
+        nmcli connection modify "$eth_connection" \
+          con-name "$eth" \
+          connection.interface-name "$eth"
     }
 
     myrepeat myrename eth0
```

I agree with @lefterisALEX but the condition should be here because the line 866 at 870 is a duplication of the myrename function without input varaible

```suggestion
    # In case of a private-only network, eth0 may not exist
    if ip link show eth0 &>/dev/null; then
      myrepeat myrename eth0
    fi
```